### ISSUE TITLE
Pairing connection test to Android Retune fails incorrectly

### ISSUE THREAD

**Issue by estreetworkshop** [ejurgensen/forked-daapd#602]
I am trying to pair my Android using Remote to forked-daapd and having problems.  In the web interface I only see "No active pairing request."

I do not see the pairing request the log /var/log/forked-daapd.log
```
[2018-10-09 01:24:37] [  LOG]     main: - Websockets
[2018-10-09 01:24:37] [  LOG]     main: - ALSA
[2018-10-09 01:24:37] [  LOG]     main: - Webinterface
[2018-10-09 01:24:37] [  LOG]     main: mDNS init
[2018-10-09 01:24:37] [  LOG]     mdns: Avahi state change: Client running
[2018-10-09 01:24:37] [  LOG]       db: Configured to use database file '/var/cache/forked-daapd/songs3.db'
[2018-10-09 01:24:37] [  LOG]       db: Now vacuuming database, this may take some time...
[2018-10-09 01:24:37] [  LOG]       db: Database OK with 2 active files and 6 active playlists
[2018-10-09 01:24:37] [  LOG]     scan: Bulk library scan completed in 0 sec
[2018-10-09 01:24:37] [  LOG]      lib: Library init scan completed in 0 sec (0 changes)
```

I can see the Remote when running avahi-browse -r -k _touch-remote._tcp
```
+   eth0 IPv4 00000000000000000000000040AC92030E85A85A      _touch-remote._tcp   local
=   eth0 IPv4 00000000000000000000000040AC92030E85A85A      _touch-remote._tcp   local
   hostname = [android-XXXX.local]
   address = [192.168.1.163]
   port = [9025]
   txt = ["RemV=10000" "txtvers=1" "DvTy=iPod" "DvNm=Moto G4" "Pair=40AC92030E85A85A" "RemN=Remote"]
```

I can telnet to the phone using the address and port listed
```
root@music:~# telnet 192.168.1.163 9025
Trying 192.168.1.163...
Connected to 192.168.1.163.
Escape character is '^]'. 
```
I am running on Ubuntu 18.04.  Any help would be very much appreciated, Thanks!

**Comment by ejurgensen** [comment]
Can you set the log level to debug and try again?

**Comment by estreetworkshop** [comment]
Yes, here is the output:

```
[2018-10-09 12:34:06] [  LOG]      lib: Library init scan completed in 0 sec (0 changes)
[2018-10-09 12:34:06] [ INFO]     mdns: Re-registering mDNS groups (services and records)
[2018-10-09 12:34:06] [ INFO]     mdns: Re-registering mDNS groups (services and records)
[2018-10-09 12:34:06] [ INFO]     mdns: Re-registering mDNS groups (services and records)
[2018-10-09 12:34:06] [ INFO]     raop: Adding AirPlay device 0023E37CB5D0@pc-erikdesktop: password: 0, verification: 0, encrypt: 1, metadata: 1, type AirPort Express 1 - 802.11g, address 192.168.1.140:5000
[2018-10-09 12:34:06] [ INFO]     raop: Adding AirPlay device 002ACB014B77@Living Room: password: 0, verification: 0, encrypt: 1, metadata: 1, type AirPort Express 1 - 802.11g, address 192.168.1.130:5000
[2018-10-09 12:34:34] [ WARN]     mdns: Connection test to 192.168.1.163:9025 failed with getsockopt return: Connection refused
[2018-10-09 12:34:34] [ WARN]     mdns: Ignoring announcement from android-158678ba545983af.local, address 192.168.1.163 is not connectable
[2018-10-09 12:34:34] [ WARN]     mdns: Connection test to 192.168.1.163:9025 failed with getsockopt return: Connection refused
[2018-10-09 12:34:34] [ WARN]     mdns: Ignoring announcement from android-158678ba545983af.local, address 192.168.1.163 is not connectable
[2018-10-09 12:35:51] [ WARN]   remote: Remote 000000000000000000000000029FBAE9E816BE91 not found in list
```

**Comment by ejurgensen** [comment]
I'm sorry to say I don't really have a clue what might be wrong if the "Connection refused" isn't true. Perhaps I can try if I can reproduce the error. What remote app is it?

**Comment by estreetworkshop** [comment]
I am the Android app Retune.

I can pair my phone successfully if I recompile commenting out line 549 in mdns_avahi.c.

**Comment by ejurgensen** [comment]
I was able to reproduce and found the cause. It was a timing problem: Retune announces the pairing service slightly before the service is actually available (the port open), which I suppose is kind of a bug in the app. Since forked-daapd's connection test is very fast, the service would not yet be available and that is why you get connection refused.

I have fixed it by generally disabling connection testing for announcements other than Airplay.
